app/src/server/db/dbOperations.ts

```diff
@@ -19,6 +19,19 @@ export async function getDraftById<T extends FhirArtifact>(id: string, resourceT
   return collection.findOne<T>({ id }, { projection: { _id: 0 } });
 }
 
+/**
+ * Retrieves the draft resource of the given type with the given url and version
+ */
+export async function getDraftByUrl<T extends FhirArtifact>(
+  url: string,
+  version: string,
```

Comment:
Happy to discuss this in more detail, but short answer is yes. This function is getting a draft artifact (from the draft artifact database) by its url. We determined that drafts must have a version; therefore, whenever we create a draft artifact, we ensure that it has a version (either by incrementing the existing artifact's version or giving it a version of 0.0.1). I am sure the version is optional in the zod validation because we used to define FhirArtifact as a Measure or Library but they should actually be a CQFMMeasure or CQFMLibrary whose version has cardinality of 1..1. Should I look into this further in this task? I am trying my best to not let things get too out of hand, so maybe we should create a task for further refining typing based on new conclusions?

Reply:
Thanks for the explanation! I forgot about the intricacies of the zod validation in regard to optional fields. I think it's fine to leave the current functionality for now.